frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts

```diff
@@ -1,16 +1,18 @@
-import { useCallback, useRef, useState } from "react";
+import { useCallback, useRef, useState, useEffect } from "react";
 import { useUnmount } from "react-use";
 
 import * as MetabaseAnalytics from "metabase/lib/analytics";
 
-import type { DashboardRefreshPeriodControls } from "../types";
+import type { DashboardRefreshPeriodControls, RefreshPeriod } from "../types";
 
 import { useInterval } from "./use-interval";
 
 const TICK_PERIOD = 1; // seconds
 export const useDashboardRefreshPeriod = ({
+  initialRefreshPeriod = null,
```

Comment:
This may not be necessary for now. I can remove this and we can talk about different options in the future

Reply:
Right now they're values defined internally by us but the idea is that the user could add one themselves. You can see the dropdown on the `Auto-refresh` button of a normal dashboard.

I was thinking the user could provide a custom value, but I've also noticed that we might not be able to do that in a nice way when it comes to the UI. We'd have to add another custom UI element to make that happen, which is why I have reservations about this one. Like - what should we do if the user provides a refresh value? do we disable changes to the refresh widget? What do we put on the dashboard? should we restrict the values to >1 min?

Just a few thoughts I had last night lol

Reply:
@oisincoveney ah, I didn't knew we also have the refresh widget in the embedded UI, TIL! then yes I would agree that removing that from the public props would be a good idea for now. let's aim to get this PR merged first 🙏🏻